test/unit/BCoWPool.t.sol

```diff
@@ -48,14 +55,78 @@ contract BCoWPool_Unit_Constructor is BaseCoWPoolTest {
 }
 
 contract BCoWPool_Unit_Finalize is BaseCoWPoolTest {
-  function test_setsApprovals(uint256 _tokensLength) public {
-    _tokensLength = bound(_tokensLength, MIN_BOUND_TOKENS, MAX_BOUND_TOKENS);
-    _setRandomTokens(_tokensLength);
-    address[] memory tokens = _getDeterministicTokenArray(_tokensLength);
-    for (uint256 i = 0; i < bPool.getNumTokens(); i++) {
+  function test_Set_Approvals() public {
+    for (uint256 i = 0; i < TOKENS_AMOUNT; i++) {
       vm.mockCall(tokens[i], abi.encodePacked(IERC20.approve.selector), abi.encode(true));
       vm.expectCall(tokens[i], abi.encodeCall(IERC20.approve, (vaultRelayer, type(uint256).max)), 1);
     }
-    bPool.finalize();
+    bCoWPool.finalize();
+  }
+}
+
+/// @notice this tests both commit and commitment
+contract BCoWPool_Unit_Commit is BaseCoWPoolTest {
+  function test_Revert_NonSolutionSettler(address sender, bytes32 orderHash) public {
+    vm.assume(sender != cowSolutionSettler);
+    vm.prank(sender);
+    vm.expectRevert(IBCoWPool.CommitOutsideOfSettlement.selector);
+    bCoWPool.commit(orderHash);
+  }
+
+  function test_Set_Commitment(bytes32 orderHash) public {
+    vm.prank(cowSolutionSettler);
+    bCoWPool.commit(orderHash);
+    assertEq(bCoWPool.commitment(), orderHash);
+  }
+}
+
+contract BCoWPool_Unit_DisableTranding is BaseCoWPoolTest {
+  function test_Revert_NonController(address sender) public {
+    // contract is deployed by this contract without any pranks
+    vm.assume(sender != address(this));
+    vm.prank(sender);
+    vm.expectRevert(IBPool.BPool_CallerIsNotController.selector);
+    bCoWPool.disableTrading();
+  }
+
+  function test_Clear_AppdataHash(bytes32 appDataHash) public {
+    vm.assume(appDataHash != bytes32(0));
+    bCoWPool.set_appDataHash(appDataHash);
+    bCoWPool.disableTrading();
+    assertEq(bCoWPool.appDataHash(), bytes32(0));
+  }
+
+  function test_Emit_TradingDisabledEvent() public {
+    vm.expectEmit();
+    emit IBCoWPool.TradingDisabled();
+    bCoWPool.disableTrading();
+  }
+
+  function test_Succeed_AlreadyZeroAppdata() public {
+    bCoWPool.set_appDataHash(bytes32(0));
+    bCoWPool.disableTrading();
+  }
+}
+
+contract BCoWPool_Unit_EnableTrading is BaseCoWPoolTest {
+  function test_Revert_NonController(address sender, bytes32 appDataHash) public {
+    // contract is deployed by this contract without any pranks
+    vm.assume(sender != address(this));
```

Comment:
I suggest using `assumeNotForgeAddress`

Reply:
IIRC that's needed when the address is going to _receive_ a call, not when it's going to be pranked, since the forge addresses have some unavoidable logic implemented in them

I checked and `address(this)` is not included in the checks of `assumeNotForgeAddress`